File: Arrays/Unsorted/Shifting/shiftingD.cpp
#include<iostream>
using namespace std;

void shiftDLeft(int arr[],int size,int d){
    int temp[d];

    //Placing first d elements to temp array
    for(int i=0;i<d;i++){
        temp[i]=arr[i];
    }

    //Shift the Remaining Elements to left
    for(int i=d;i<size;i++){
        arr[i-d]=arr[i];
    }

    //Place the elements of temp array to original array
    for(int i=0;i<d;i++){
        arr[size-d+i]=temp[i];
    }
}

void printArray(int arr[],int size){
    for(int i=0;i<size;i++){
        cout<<arr[i]<<" ";
    }
    cout<<"\n";
}

int main(){
    int arr[]={5,4,3,2,1};
    int size=sizeof(arr)/sizeof(arr[0]);
    cout<<"Array Before Shifting\n";
    printArray(arr,size);

    shiftDLeft(arr,size,2);

    cout<<"Array After Shifting\n";
    printArray(arr,size);
}

File: Arrays/Unsorted/Shifting/shiftingD.js
const shiftDLeft=(arr,d)=>{
    let temp=[]
    
    //place the First d elements in temporary array
    for(let i=0;i<d;i++){
        temp.push(arr[i])
    }

    //Shift the remaining elements to left
    for(let i=d;i<arr.length;i++){
       arr[i-d]=arr[i];
    }

    //Place the Initially Shifted Elements
    for(let i=0;i<d;i++){
        arr[i+arr.length-d]=temp[i];
    }
}

let arr=[1,2,3,4,5]

console.log("Before Shifting")
for(let i=0;i<arr.length;i++){
   console.log(arr[i])
}

shiftDLeft(arr,3)

console.log("After Shifting")
for(let i=0;i<arr.length;i++){
   console.log(arr[i])
}
